docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,404 @@
+
+    class MyTile(
+      val myParams: MyTileParams,
+      crossing: ClockCrossingType,
+      lookup: LookupByHartIdImpl,
+      q: Parameters,
+      logicalTreeNode: LogicalTreeNode)
+      extends BaseTile(myParams, crossing, lookup, q)
+      with SinksExternalInterrupts
+      with SourcesExternalNotifications
+    {
+
+      // Private constructor ensures altered LazyModule.p is used implicitly
+      def this(params: MyTileParams, crossing: RocketCrossingParams, lookup: LookupByHartIdImpl, logicalTreeNode: LogicalTreeNode)(implicit p: Parameters) =
+        this(params, crossing.crossingType, lookup, p, logicalTreeNode)
+
+      // Require TileLink nodes
+      val intOutwardNode = IntIdentityNode()
+      val masterNode = visibilityNode
+      val slaveNode = TLIdentityNode()
+
+      // Implementation class (See below)
+      override lazy val module = new MyTileModuleImp(this)
+
+      // Required entry of CPU device in the device tree for interrupt purpose
+      val cpuDevice: SimpleDevice = new SimpleDevice("cpu", Seq("my-organization,my-cpu", "riscv")) {
+        override def parent = Some(ResourceAnchors.cpus)
+        override def describe(resources: ResourceBindings): Description = {
+          val Description(name, mapping) = super.describe(resources)
+          Description(name, mapping ++
+                            cpuProperties ++
+                            nextLevelCacheProperty ++
+                            tileProperties)
+        }
+      }
+
+      ResourceBinding {
+        Resource(cpuDevice, "reg").bind(ResourceAddress(hartId))
+      }
+
+      // (Connection to bus, interrupt, etc.)
+    }
+
+TileLink Connection
+-------------------
+
+Chipyard use TileLink as its onboard bus protocol, and if your core doesn't use TileLink, you will need to convert them
+in the tile class. Below is an example of how to connect a core using AXI4 to the TileLink bus: 
+
+.. code-block:: scala
+
+    val memoryTap = TLIdentityNode() // Every bus connection should have their own tap node
+    (tlMasterXbar.node  // tlMasterXbar is the bus crossbar to be used when this core / tile is acting as a master; otherwise, use tlSlaveXBar
+      := memoryTap
+      := TLBuffer()
+      := TLFIFOFixer(TLFIFOFixer.all) // fix FIFO ordering
+      := TLWidthWidget(beatBytes) // reduce size of TL
+      := AXI4ToTL() // convert to TL
+      := AXI4UserYanker(Some(2)) // remove user field on AXI interface. need but in reality user intf. not needed
+      := AXI4Fragmenter() // deal with multi-beat xacts
+      := memAXI4Node) // The custom node, see below
+
+Remember, you may not need all of these intermediate widgets. See :ref:`diplomatic_widgets` for the meaning of each intermediate
+widget. If you are using TileLink, then you only need the tap node and the TileLink node used by your components. Also, Chipyard
+support AHB, APB and AXIS, and most of the AXI4 widgets has equivalent widget for these bus protocol. See the reference page for
```

Comment:
I would be more precise here, and say that RocketChip provides converters between various protocols and TileLink

Reply:
Done.